Api/Client.php

```diff
@@ -200,13 +199,15 @@ private function getFilesFromReportRequest(ReportRequest $reportRequest, $name,
         $reportRequestId = $this->submitGenerateReport($reportRequest, $name);
         $reportRequestStatus = $this->waitForStatus($reportRequestId);
         $reportDownloadUrl = $reportRequestStatus->ReportDownloadUrl;
-        $zipFile = $this->fileHelper->getFile($reportDownloadUrl, $downloadFile);
-        if ($zipFile !== false) {
-            $this->files = $this->fileHelper->unZip($zipFile);
-            $this->fixFile($report);
+        $file = $this->fileHelper->copyFile($reportDownloadUrl, $downloadFile);
+
+        if ($this->fileHelper->isHealthyZipFile($file)) {
+            $files = $this->fixFile($report, $this->fileHelper->unZip($file));
+        } else {
+            $files = $file;
```

Comment:
Can you always make it an array of files then the check is_array is not needed anymore